sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -205,6 +206,12 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
       }
     }
 
+    private def hashJoinSupported
+        (leftKeys: Seq[Expression], rightKeys: Seq[Expression]): Boolean = {
+      leftKeys.forall(e => UnsafeRowUtils.isBinaryStable(e.dataType)) &&
```

Comment:
```suggestion
      leftKeys.union(rightKeys).forall(e => UnsafeRowUtils.isBinaryStable(e.dataType))
      }
```

Just a suggestion -- Whichever you prefer. Doesn't actually matter.

Reply:
Went with `leftKeys.concat(rightKeys)` since IDE complained that `union` is deprecated.